src/pages/Configurator/WorkoutEditor/WorkoutEditor.tsx

```diff
@@ -1,22 +1,46 @@
-import React from 'react';
+import React, { FunctionComponent } from 'react';
 import { observer } from 'mobx-react-lite';
-import { NavigationStackScreenProps } from 'react-navigation-stack';
+import { RouteProp, CompositeNavigationProp } from '@react-navigation/native';
+import { BottomTabNavigationProp } from '@react-navigation/bottom-tabs';
+import { StackNavigationProp } from '@react-navigation/stack';
 import styled from '@woap/utils/styled-components';
 import { ActionButton } from '@woap/components/ActionButton';
 import { ExercisesList } from '@woap/components/ExercisesList';
 import { InputTitle } from '@woap/components/InputTitle';
 import { Routes } from '@woap/navigation/routes';
+import { WorkoutsNavigatorParamList } from '@woap/navigation/WorkoutsNavigator';
+import { TabNavigatorParamList } from '@woap/navigation/TabNavigator';
+import { RootNavigatorParamList } from '@woap/navigation';
 
-export const WorkoutEditor = observer(({ navigation }: NavigationStackScreenProps) => {
-  const workout = navigation.getParam('workout');
+type WorkoutEditorScreenNavigationProp = CompositeNavigationProp<
+  StackNavigationProp<RootNavigatorParamList, Routes.TabNavigator>,
+  CompositeNavigationProp<
+    BottomTabNavigationProp<TabNavigatorParamList, Routes.WorkoutsNavigator>,
+    StackNavigationProp<WorkoutsNavigatorParamList, Routes.WorkoutEditor>
+  >
+>;
+
+type WorkoutEditorScreenRouteProp = RouteProp<WorkoutsNavigatorParamList, Routes.WorkoutEditor>;
+
+type Props = {
+  navigation: WorkoutEditorScreenNavigationProp;
+  route: WorkoutEditorScreenRouteProp;
+};
+
+export const WorkoutEditor: FunctionComponent<Props> = observer(({ navigation, route }) => {
+  const workout = route.params.workout;
+  const setWorkoutName = (text: string) => workout.setName(text);
```

Comment:
why the refacto ?

Reply:
correction warning de perte de binding du this